src/dhcpd/main.rs

```diff
@@ -20,20 +21,71 @@ macro_rules! try_fmt {
     )
 }
 
-fn dhcp(quiet: bool) -> Result<(), String> {
-    let current_mac = MacAddr::from_str(try_fmt!(getcfg("mac"), "failed to get current mac").trim());
+fn get_cfg_value(path: &str) -> Result<String, String> {
+    let path = format!("netcfg:{}", path);
+    let mut file = File::open(&path).map_err(|_| format!("Can't open {}", &path))?;
+    let mut result = String::new();
+    file.read_to_string(&mut result)
+        .map_err(|_| format!("Can't read {}", path))?;
+    Ok(result)
+}
+
+fn get_iface_cfg_value(iface: &str, cfg: &str) -> Result<String, String> {
+    let path = format!("ifaces/{}/{}", iface, cfg);
+    get_cfg_value(&path)
+}
+
+fn set_cfg_value(path: &str, value: &str) -> Result<(), String> {
+    let path = format!("netcfg:{}", path);
+    let mut file = OpenOptions::new().read(false).write(true).create(false).open(&path)
+        .map_err(|_| format!("Can't open {}", path))?;
+    file.write(value.as_bytes())
+        .map(|_| ())
+        .map_err(|_| format!("Can't write {} to {}", value, path))?;
+    file.sync_data()
+        .map_err(|_| format!("Can't commit {} to {}", value, path))
+}
+
+fn set_iface_cfg_value(iface: &str, cfg: &str, value: &str) -> Result<(), String> {
+    let path = format!("ifaces/{}/{}", iface, cfg);
+    set_cfg_value(&path, value)
+}
 
-    let current_ip = getcfg("ip").unwrap_or("0.0.0.0".to_string());
-    if ! quiet {
-        println!("DHCP: MAC: {} Current IP: {}", current_mac.to_string(), current_ip.trim());
+fn dhcp(iface: &str, quiet: bool) -> Result<(), String> {
+    let current_mac = MacAddr::from_str(get_iface_cfg_value(iface, "mac")?.trim());
+
+    let current_ip = get_iface_cfg_value(iface, "addr/list")?
+        .lines()
+        .next()
+        .map(|l| l.to_owned())
+        .unwrap_or("0.0.0.0".to_string());
```

Comment:
Do we need to filter out loopback addrs here?

Reply:
The loopback should always be the last address in the list, otherwise, the network won't work. `netcfg` tries make it so by putting new addrs at the top of the list, but overall these are ad-hoc hacks until we have a proper loopback interface.

Reply:
:+1: